main.go

```diff
@@ -162,47 +135,120 @@ func main() {
+	switch cacheLevel {
+	case cacheInputNone:
+		return cache.Cache{}
+	case cacheInputlocal:
+		localCaches, err := collectLocalCaches(basePth)
+		if err != nil {
+			log.Warnf("Error occurred while getting local cache. Error: %s", err)
+		}
+		for _, lcItem := range localCaches {
+			nuGetCache.IncludePath(lcItem)
+		}
+	case cacheInputGlobal:
+		nuGetCache.IncludePath(collectGlobalCaches())
+	case cacheInputAll:
+		localCaches, err := collectLocalCaches(basePth)
+		if err != nil {
+			log.Warnf("Error occurred while getting all cache. Error: %s", err)
+		}
+		for _, lcItem := range localCaches {
+			nuGetCache.IncludePath(lcItem)
+		}
+		nuGetCache.IncludePath(collectGlobalCaches())
+	}
+	return nuGetCache
+}
 
-	nugetCache := cache.New()
+// collectHTTPCaches collects the HTTP cache.
+func collectHTTPCaches() (string, error) {
+	httpCachePth := HTTPCachePath()
+	if exists, err := pathutil.IsPathExists(httpCachePth); err != nil {
+		return "", fmt.Errorf("failed to determine if path (%s) exists, error: %s", httpCachePth, err)
+	} else if exists {
+		return httpCachePth, nil
+	}
+	return "", nil
+}
 
-	xamarinHomeCache := filepath.Join(pathutil.UserHomeDir(), ".local", "share", "Xamarin")
+// HTTPCachePath gets the path for the HTTP cache.
+func HTTPCachePath() string {
+	if pth := os.Getenv(cacheEnvHTTP); pth != "" {
+		return pth
+	}
+	return filepath.Join(pathutil.UserHomeDir(), ".local", "share", "NuGet", "v3-cache")
+}
 
-	if exists, err := pathutil.IsPathExists(xamarinHomeCache); err != nil {
-		log.Warnf("Failed to determine if path (%s) exists, error: %s", xamarinHomeCache, err)
-	} else if exists {
-		nugetCache.IncludePath(xamarinHomeCache)
+// collectGlobalCaches collects the global package caches.
+func collectGlobalCaches() string {
+	if pth := os.Getenv(cacheEnvGlobal); pth != "" {
+		return pth
 	}
+	return filepath.Join(pathutil.UserHomeDir(), ".nuget", "packages")
+}
 
-	absProjectRoot, err := filepath.Abs(".")
+// collectLocalCaches collects the local caches.
+func collectLocalCaches(basePth string) ([]string, error) {
+	var caches []string
+	absProjectRoot, err := filepath.Abs(basePth)
 	if err != nil {
-		log.Warnf("Cache collection skipped: failed to determine project root path.")
-	} else {
-		err := filepath.Walk(absProjectRoot, func(path string, f os.FileInfo, err error) error {
-			if f.IsDir() {
-				if f.Name() == "packages" {
-					nugetCache.IncludePath(path)
-					return io.EOF
-				}
+		return []string{}, fmt.Errorf("cache collection skipped: failed to determine project root path")
+	}
+	if err := filepath.Walk(absProjectRoot, func(path string, f os.FileInfo, err error) error {
+		if f.IsDir() {
+			if f.Name() == "packages" {
+				caches = append(caches, path)
+				return io.EOF
 			}
-			return nil
-		})
-		if err == io.EOF {
-			err = nil
 		}
+		return nil
+	}); err != nil && err != io.EOF {
+		return []string{}, fmt.Errorf("cache collection skipped: failed to determine cache paths. Error: %s", err)
```

Comment:
pls update `cache collection skipped: failed to determine cache paths: %s` : https://golang.org/doc/effective_go.html#errors

Reply:
pls apply it everywhere you log/compose an error